TerrariaServerAPI/TerrariaApi.Server/Hooking/ItemHooks.cs

```diff
@@ -15,36 +15,34 @@ public static void AttachTo(HookManager hookManager)
 		{
 			_hookManager = hookManager;
 
-			Hooks.Item.PreSetDefaultsById = OnPreSetDefaultsById;
-			Hooks.Item.PreNetDefaults = OnPreNetDefaults;
-			Hooks.Chest.QuickStack = OnQuickStack;
+			On.Terraria.Item.SetDefaults_int_bool += OnSetDefaults;
+			On.Terraria.Item.netDefaults += OnNetDefaults;
+
+			Hooks.Chest.QuickStack += OnQuickStack;
 		}
 
-		static HookResult OnPreSetDefaultsById(Item item, ref int type, ref bool noMatCheck)
+		private static void OnNetDefaults(On.Terraria.Item.orig_netDefaults orig, Item self, int type)
 		{
-			if (_hookManager.InvokeItemSetDefaultsInt(ref type, item))
-			{
-				return HookResult.Cancel;
-			}
-			return HookResult.Continue;
+			if (_hookManager.InvokeItemNetDefaults(ref type, self))
+				return;
+
+			orig(self, type);
 		}
 
-		static HookResult OnPreNetDefaults(Item item, ref int type)
+		private static void OnSetDefaults(On.Terraria.Item.orig_SetDefaults_int_bool orig, Item self, int Type, bool noMatCheck)
```

Comment:
It isn't that intuitive to know what `On.Terraria.Item.orig_SetDefaults_int_bool orig` is here. This is the original method, right?

Reply:
these are auto-generated monomod hooks, in fact most of the OTAPI.Runtime.dll consist of these, which are generated using the vanilla methods signature to make the type unique.
to use these events you should already know what the method is for essentially, as its working low level with the game engine.

Reply:
Yeah that’s fair. I guess we can put a note somewhere about it, but nbd